PyStemmusScope/save.py

```diff
@@ -0,0 +1,334 @@
+    if var_name == "SoilTemp":
+        # Celsius to Kelvin : K = 273.15 + C
+        data = data + 273.15
+
+    elif var_name == "SoilMoist":
+        # cm to m
+        thicknesses = np.array(thicknesses) / 100.0
+
+        for index in data.index:
+            # m3/m3 to kg/m2
+            volumetric_water_content = np.array(data.loc[index])
+            data.loc[index] = vc.soil_moisture(volumetric_water_content, thicknesses)
+
+    # reshape the data frame
+    data = data.stack()
+
+    # set values
+    layers = range(1, data.index.levshape[1] + 1)
+    data.index.names = ["time", "z"]
+    data.index = data.index.set_levels([time, layers], level=["time", "z"])
+    data.name = var_name
+
+    # convert dataframe to xarray data array
+    data_array = data.to_xarray()
+
+    # add z attributes
+    data_array["z"].attrs = {
+        "long_name": "Soil layer",
+        "standard_name": "Soil layer",
+        "definition": soil_metadata,
+        "units": "-",
+    }
+
+    return data_array
+
+
+def _prepare_3d_data(file_name: str, model_name: str, alma_name: str, time: List) ->  xr.DataArray:
+    """Reshape a `pandas.DataFrame` to include z dimension.
+
+    Args:
+        file_name(str): csv file name generated by Stemmus_Scope model.
+        model_name(str): variable name by Stemmus_Scope model.
+        alma_name(str): variable name by ALMA conventions.
+        time(list): time values to be used for the time coordinates.
+
+    Returns:
+        xr.DataArray: a dataarray with one dimension of time.
+    """
+    # the first three rows are names and units
+    data = pd.read_csv(file_name, delimiter=",")
+
+    # select variable and skip first row that is unit
+    data = data[model_name].iloc[1:]
+
+    # set time values
+    data.index = time
+    data.index.names = ["time"]
+
+    # rename it to alma name
+    data.name = alma_name
+
+    # make sure it is float and not str
+    data = data.astype('float32')
+
+    # convert dataframe to xarray data array
+    return data.to_xarray()
+
+
+def _create_soil_layer_metadata(thicknesses: List, depths: List) -> List:
+    """
+    layer_1: 0.0 - 1.0 cm
+    layer_2: 1.0 - 2.0 cm
+    layer_3: 2.0 - 3.0 cm
+    """
+
+    metadata = []
+    for index, (thickness, depth) in enumerate(zip(thicknesses, depths)):
+        metadata.append(f"layer_{index + 1}: {(depth - thickness)} - {depth} cm")
+
+    return metadata
+
+
+def _update_dataset_attrs_dims(dataset: xr.Dataset, forcing_dict: Dict) -> xr.Dataset:
+    """Update dimentions of a dataset according to ALMA conventions.
+
+    Args:
+        dataset(xr.Dataset): a dataset with varaibles in ALMA conventions.
+
+    Returns:
+        xr.Dataset: the dataset with dimensions ("time", "x", "y").
+    """
+
+    # add x/y dims to the dataset
+    dataset_expanded = dataset.expand_dims(["x", "y"])
+
+    # change the order of dims
+    try:
+        dataset_reordered = dataset_expanded.transpose("time", "y", "x", "z")
+    except ValueError:
```

Comment:
I amnot sure about the changes you suggested here. but I refactored the code, see my [commit](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/pull/23/commits/3dadf8fc89790276d2b95db524af0c9333949402).

Reply:
`try`, `except` statements should generally be avoided in cases like this.

I'd suggest you refactor this to the following (or similar) to avoid `try`:
```python
req_dims = ['time', 'x', 'y']
if not all(dim in dataset_expanded.dims for dim in req_dims):
    raise ValueError("Data should have dimensions 'time', 'y', 'z'.")
elif 'z' in dataset_expanded.dims:
    dataset_reordered = dataset_expanded.transpose("time", "y", "x", "z")
else:
    dataset_reordered = dataset_expanded.transpose("time", "y", "x")
```

Reply:
thanks for the suggestion. I refactored it, see my [commit](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/pull/23/commits/90d21e6f66eaa0723146e1eb90ada7a76c246021).